### PR TITLE
Feature/databricks vector search

### PR DESCRIPTION
**Pull request by NickhilN**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by hatianzhang** [comment]
@NickhilN thanks. could you help fix those lint errors.

**Review by logan-markewich — COMMENTED** [review]
@NickhilN Can you add some example notebook in `docs/examples/vector_stores` ? An link to it in `docs/module_guides/storing/vector_stores.md`

**Comment by logan-markewich** [comment]
@NickhilN are you able to address the above?

File: llama-index-integrations/vector_stores/llama-index-vector-stores-databricks-vector-search/llama_index/vector_stores/databricks-vector-search/base.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        self._index.delete(
+            primary_keys=[ref_doc_id],
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-databricks-vector-search/llama_index/vector_stores/databricks-vector-search/base.py:276]: This isn't actually correct. The intention here is to delete any node with `doc_id == ref_doc_id` in its metadata. Is this possible here?

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**